proto/stream_plan.proto

```diff
@@ -429,11 +429,9 @@ message DynamicFilterNode {
   catalog.Table left_table = 3;
   // Right table stores single value from RHS of predicate.
   catalog.Table right_table = 4;
-  // It is true when the right side of the inequality predicate is monotonically:
-  // - decreasing for <, <=, increasing for >, >=
-  // bool is_monotonic = 10;
```

Comment:
What about case where we have non-monotonic input. 🤔 

For now I suppose it's not supported. But I guess that's the intention of having the `is_monotonic` flag previously.

Perhaps we should just keep that field commented out instead of removing it.

I don't think `condition_always_relax` can handle it.
Because when `condition_always_relax` is false, I think the current intention is the make it mean that right hand's side change always more the condition more constrained.

However, the opposite of it actually implies that right side's change sometimes makes the condition more relaxed, sometimes more constrained.

Reply:
when we need it, we can add it back. I think the field should be on-demand by the executor with its specific optimization.

Reply:
In that case I think we should still add some comment, because `DynamicFilter`'s RHS may not always be monotonic.

So for the `condition_always_relax` we should add a comment which says that we assume the input to always be monotonically changing.